### PR TITLE
basic powerspectra code + example

### PR DESCRIPTION
**Pull request by bhorowitz**
A little tf implementation of a powerspectra estimator (more or less copied from nbodykit's projected PS implementation)... It seems to match that from nbodykit in the couple realizations/boxsizes I chose. It currently doesn't go up to as high of k as nbodykit by default; probably something to do with nyquist frequency definitions.

The IO for the pk function is very rough; depending on how we end up defining other functions perhaps there could be some standardization such as wrapping up all ps-related functions (sigma8? HMF?) in some class.

### CODE REVIEW

File: code/utils/ps.py
Comment (bhorowitz) [inline code/utils/ps.py]: probably should go to higher k here... not sure if we want to have a kmax or similar option here...

File: code/utils/ps.py
Comment (bhorowitz) [inline code/utils/ps.py]: I don't think these really have to be tf.Variables, but things work if they are tf variables? *shrug*

Comment (bhorowitz) [reply code/utils/ps.py]: maybe I could spend like 5 minutes to make this whole part differentiable too in case someone ever wants to take the derivative with respect to box size :P

**Comment by EiffL** [comment]
I've looked at this PR, it looks pretty good :-) I'll just wait until we have code to do the CIC painting, so that we can test computing the  powerspectrum directly from a TF field.
This is going to happen with #18

**Comment by EiffL** [comment]
Ok, so I've used this stuff in this demo notebook: 
https://github.com/DifferentiableUniverseInitiative/DHOD/blob/u/EiffL/FlowPM/nb/PowerSpectrum.ipynb 

It seems to work :-) but I have a few comments:
  - It seems that the code does work if my box has nc=128, and bs=250, there are bins with Num =0 in it, so that it returns NaNs. It's essentially I think just a matter of not creating a k vector too long, but I haven't worked it through....

  - Could you move only the pk code (and associated function) to a file in  
```
diffhod/mock_observables/power_spectrum.py
```
This way, it follows a structure similar to halotools, and then people would use with:
```
from diffhod.mock_observables import pk
... = pk(....)
```

**Comment by bhorowitz** [comment]
Okay! I fixed the kmax issue, and added a very trivial test (pk of noise) which I compared vs. nbodykit's powerspectra. I wasn't sure if we wanted to add a requirement for flowpm in order to do a slightly more nuanced test involving a flowpm mesh object with a real density field.

**Comment by EiffL** [comment]
Awesome :-D I'm happy to leave flowpm out of the validation of the power spectrum. waiting to see if we get the green light from github CI :-)

**Comment by EiffL** [comment]
Ok fantatstic! Let me just try to massage the PR into a cleaner version....

**Comment by EiffL** [comment]
@bhorowitz here is the magic I used to clean up the PR and only show your modifications in the list of changed files:
```
$ git checkout diffps
$ git rebase origin/master
$ git push -f
```
That simplifies the git history, and only keeps the modifications that differ from the current master branch. More info here: https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request#perform-a-rebase

**Review by EiffL — APPROVED** [review]
LGTM

**Comment by EiffL** [comment]
Ok Ben, it's ok for me if you want to press the big Green Merge Button, go ahead ;-) 
And in the meantime, @all-contributors please add @bhorowitz for code

**Pull request merged**
